core/src/test/java/com/datastrato/gravitino/storage/TestEntityStorage.java

```diff
@@ -0,0 +1,1447 @@
+  public static final String KV_STORE_PATH =
+      "/tmp/gravitino_kv_entityStore_" + UUID.randomUUID().toString().replace("-", "");
+
+  private static final String JDBC_STORE_PATH =
+      "/tmp/gravitino_jdbc_entityStore_" + UUID.randomUUID().toString().replace("-", "");
+  private static final String DB_DIR = JDBC_STORE_PATH + "/testdb";
+
+  static Object[] storageProvider() {
+    return new Object[] {Configs.DEFAULT_ENTITY_STORE, Configs.RELATIONAL_ENTITY_STORE};
+  }
+
+  private void init(String type, Config config) {
+    Preconditions.checkArgument(StringUtils.isNotBlank(type));
+    if (type.equals(Configs.DEFAULT_ENTITY_STORE)) {
+      try {
+        FileUtils.deleteDirectory(FileUtils.getFile(KV_STORE_PATH));
+      } catch (Exception e) {
+        // Ignore
+      }
+      Mockito.when(config.get(ENTITY_STORE)).thenReturn("kv");
+      Mockito.when(config.get(ENTITY_KV_STORE)).thenReturn(DEFAULT_ENTITY_KV_STORE);
+      Mockito.when(config.get(Configs.ENTITY_SERDE)).thenReturn("proto");
+      Mockito.when(config.get(ENTRY_KV_ROCKSDB_BACKEND_PATH)).thenReturn(KV_STORE_PATH);
+
+      Assertions.assertEquals(KV_STORE_PATH, config.get(ENTRY_KV_ROCKSDB_BACKEND_PATH));
+      Mockito.when(config.get(STORE_TRANSACTION_MAX_SKEW_TIME)).thenReturn(1000L);
+      Mockito.when(config.get(KV_DELETE_AFTER_TIME)).thenReturn(20 * 60 * 1000L);
+    } else if (type.equals(Configs.RELATIONAL_ENTITY_STORE)) {
+      File dir = new File(DB_DIR);
+      if (dir.exists() || !dir.isDirectory()) {
+        dir.delete();
+      }
+      dir.mkdirs();
+      Mockito.when(config.get(ENTITY_STORE)).thenReturn(RELATIONAL_ENTITY_STORE);
+      Mockito.when(config.get(ENTITY_RELATIONAL_STORE)).thenReturn(DEFAULT_ENTITY_RELATIONAL_STORE);
+      Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_URL))
+          .thenReturn(String.format("jdbc:h2:%s;DB_CLOSE_DELAY=-1;MODE=MYSQL", DB_DIR));
+      Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_USER)).thenReturn("root");
+      Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_PASSWORD)).thenReturn("123");
+      Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_DRIVER)).thenReturn("org.h2.Driver");
+    } else {
+      throw new UnsupportedOperationException("Unsupported entity store type: " + type);
+    }
+  }
+
+  private void prepareJdbcTable() {
+    // Read the ddl sql to create table
+    String scriptPath = "h2/h2-init.sql";
+    try (SqlSession sqlSession =
+            SqlSessionFactoryHelper.getInstance().getSqlSessionFactory().openSession(true);
+        Connection connection = sqlSession.getConnection();
+        Statement statement = connection.createStatement()) {
+      URL scriptUrl = ClassLoader.getSystemResource(scriptPath);
+      if (scriptUrl == null) {
+        throw new IllegalStateException("Cannot find init sql script:" + scriptPath);
+      }
+      StringBuilder ddlBuilder = new StringBuilder();
+      try (InputStreamReader inputStreamReader =
```

Comment:
IOUtils.readLines(this.getClass().getClassLoader().getResourceAsStream("xxxx"), StandardCharsets.UTF_8);

Reply:
Done